bin/client/main.rs

```diff
@@ -246,8 +246,21 @@ async fn run() -> std::result::Result<(), ClientError> {
                 None => ensure_config_dir().await?.join("secret_keys.toml"),
             };
 
-            // Signing key
-            let key = first_matching_key(keyfile, &role).await?;
+            let keys = SecretKeyFile::load_file(&keyfile).await.map_err(|e| {
+                ClientError::Other(format!("Error loading file {}: {}", keyfile.display(), e))
+            })?;
+
+            let match_type = if sign_opts.label.is_some() {
```

Comment:
I really like the enum way of doing this! 

One nit here: Probably better to do an `if let Some(label) = sign_opts.label.as_ref()` instead

Reply:
Used Match now, as we are matching based on 2 expressions, using `if let` I felt is a bit strange !!
https://github.com/VishnuJin/bindle/blob/e9e0100b3cca0f5301649f6e5d213fd9aab54f56/bin/client/main.rs#L253-L260